server/icat/src/icatHighLevelRoutines.cpp

```diff
@@ -89,11 +90,11 @@ int chlOpen() {
     // cache the database type for subsequent calls
     try {
         const auto& database_plugin_map = irods::get_server_property<const nlohmann::json&>(std::vector<std::string>{irods::KW_CFG_PLUGIN_CONFIGURATION, irods::KW_CFG_PLUGIN_TYPE_DATABASE});
-        if ( database_plugin_map.size() != 1 ) {
+        if (database_plugin_map.empty()) {
             rodsLog( LOG_ERROR, "Database plugin map must contain exactly one plugin object" );
             return SYS_INVALID_INPUT_PARAM;
         }
-        database_plugin_type = database_plugin_map.items().begin().key();
+        database_plugin_type = database_plugin_map.at(irods::KW_CFG_DB_TECHNOLOGY).get_ref<const std::string&>();
     } catch ( const irods::exception& e ) {
```

Comment:
Let's cover the non-irods exception case too.

Reply:
Done.